app/controllers/api/subcollections/container_images.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Api
+  module Subcollections
+    module ContainerImages
+      def container_images_scan_resource(provider_id, type, image_id, _payload)
+        api_action(type, image_id, :skip_href => true) do |klass|
+          image = resource_search(image_id, type, klass)
+          begin
+            desc = "#{container_image_ident(image)} scanning"
+            task_id = image.scan
+            action_result(true, desc).merge(:task_id => task_id, :parent_id => provider_id)
+          rescue => err
+            action_result(false, err.to_s)
```

Comment:
Do you know what conditions might result in an error? If so, can you add a test for that case?

Reply:
I am not sure, this is basically creating a job and the rescue is here for unforeseen errors. I think we can use `allow(image).to receive(:scan) ....` magic to mimic this. wdy?